google/cloud/internal/external_account_token_source_aws_test.cc

```diff
@@ -103,6 +114,159 @@ ExternalAccountTokenSourceAwsInfo MakeTestInfoImdsV2() {
   return info;
 }
 
+TEST(ExternalAccountTokenSource, SourceWorking) {
+  // To simplify the test we provide all the parameters via environment
+  // variables and avoid using imdsv2.
+  auto const region = ScopedEnvironment("AWS_REGION", "expected-region");
+  auto const access_key_id =
+      ScopedEnvironment("AWS_ACCESS_KEY_ID", "test-access-key-id");
+  auto const secret_access_key =
+      ScopedEnvironment("AWS_SECRET_ACCESS_KEY", "test-secret-access-key");
+  auto const token =
+      ScopedEnvironment("AWS_SESSION_TOKEN", "test-session-token");
+  auto const credentials_source = nlohmann::json{
+      {"environment_id", "aws1"},
+      {"region_url", kTestRegionUrl},
+      {"url", kTestMetadataUrl},
+      {"regional_cred_verification_url", kTestVerificationUrl},
+      // {"imdsv2_session_token_url", kTestImdsv2Url},
+  };
+  auto const target = std::string{
+      "//iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/"
+      "workloadIdentityPools/$POOL_ID/providers/$PROVIDER_ID"};
+  auto const source = MakeExternalAccountTokenSourceAws(
+      credentials_source, target, MakeTestErrorContext());
+  ASSERT_STATUS_OK(source);
+
+  MockClientFactory client_factory;
+  EXPECT_CALL(client_factory, Call).Times(0);
+
+  auto subject_token = (*source)(client_factory.AsStdFunction(), Options{});
+  ASSERT_STATUS_OK(subject_token);
+  // Verify some simple cases. We make a full test in ComputeSubjectToken().
+  EXPECT_THAT(subject_token->token, HasSubstr(nlohmann::json{
+                                        {"key", "x-goog-cloud-target-resource"},
+                                        {"value",
+                                         target}}.dump()));
+  EXPECT_THAT(
+      subject_token->token,
+      HasSubstr(nlohmann::json{{"key", "host"},
+                               {"value", "sts.expected-region.amazonaws.com"}}
+                    .dump()));
+}
+
+TEST(ExternalAccountTokenSource, SourceImdsv2Failure) {
+  // To simplify the test we provide all the parameters via environment
+  // variables and avoid using imdsv2.
```

Comment:
This test seems to use imdsv2. Should we remove the comment and not set any env vars?